crates/ruma-common/tests/api/status_override.rs

```diff
@@ -0,0 +1,52 @@
+#![allow(clippy::exhaustive_structs)]
+
+use http::{header::{Entry, LOCATION}, StatusCode};
+use ruma_common::{
+    api::{
+        request, response, Metadata, OutgoingResponse as _,
+    },
+    metadata,
+};
+
+const METADATA: Metadata = metadata! {
+    method: GET,
+    rate_limited: false,
+    authentication: None,
+    history: {
+        unstable => "/_matrix/my/endpoint",
+    }
+};
+
+/// Request type for the `status_override` endpoint.
+#[request]
+pub struct Request {}
+
+/// Response type for the `status_override` endpoint.
+#[response(status = FOUND)]
+pub struct Response {
+    #[ruma_api(header = LOCATION)]
+    pub location: Option<String>,
+}
+
+#[test]
+fn response_status_override() {
```

Comment:
It would be great if there was also a test somewhere that checks the default status code.